UoaWDCC/patrols
# email api

## Context

Integrated email API into the backend, using the service "Resend".

## What Changed?

- Added the emailController.ts 
- Added email.ts 
- Added Resend package

## How To Review

1. api/rotes/email.ts
2. api/controller/reportController.ts

## Testing

1. Go to https://resend.com and create an account or log in via GitHub.
2. Go to API Keys on the sidebar
3. Create a new API key
4. Add the API KEY as an environment variable in .env file. 
5. cd to api in the terminal
6. run  "yarn dev"
7. Use the link below to test
https://www.postman.com/jmak320/workspace/cpnz-reporttesting/request/34420046-533cd425-acc5-4702-978e-bbe85111754c?action=share&source=copy-link&creator=34420046&ctx=documentation

In Postman:

email: Use the email you have registered with Resend in the email field
patrolName: Any name
PatrolID: Any Integer
formData: The content you wish to put in the email

8. Check your email to see if it has been sent successfully. (The sender address: onboarding@resend.dev)

## Risks

Spamming the Send key on Postman could result in an overwhelmingly huge amount of emails 

## Notes

- Due to not having a verified domain currently. Resend only allows us to send email to the email address that the API Key is registered with.

- Using Resend allows Front end using React to compose the email format.
